### DISCUSSION TITLE
what is the best architecture choice for many cameras?

### DISCUSSION THREAD

**Discussion by throberto** [bluenviron/mediamtx#889] [Questions] [answered]
Hey, guys.
I need to deal with an IoT project where I have distributed IP cameras, where each client will have between 10 and 20 cameras producing images all the time.

I tried to play the images directly to AWS Kinesis Video Stream but for some reason I was only able to through this RTSP server, following [this](https://github.com/aler9/rtsp-simple-server/issues/464) problem.

My question is in the limits of the rtsp server in relation to the amount of live cameras, and what would be the best approach to communicate with the cameras inside the aws in this use case.

<img width="421" alt="image" src="https://user-images.githubusercontent.com/56898810/159067760-f95fbb57-4dfb-43c2-9849-d939dc444edc.png">

My goal is to stream to AWS Kinesis Video Streams from on-premises cameras (many).

After all, what would be the best approach in your opinion?

**Comment by aler9** [comment, marked as answer]
Hello,

The first option is unfeasible since it would require direct access to the cameras from the internet, and that's not a good idea. If you still want to pursue it, my advice is to use a VPN to link the EC2 instance with the cameras. It requires a Raspberry PI with a VPN client on it.

The second option is slightly better, since only RTSPSS is exposed on the internet - a VPN is recommended anyway.

> My question is in the limits of the rtsp server in relation to the amount of live cameras

I'm handling right now a rtsp-simple-server instance that proxies 150 cameras. it consumes 2.7 GB of RAM and 4.5 CPUs. Computational power is generally not a problem.

[a comment by an automated account is not shown]
